File: Server/SuperBoard/src/main/java/com/narara/superboard/reply/service/ReplyService.java
package com.narara.superboard.reply.service;

import com.google.firebase.messaging.FirebaseMessagingException;
import com.narara.superboard.member.entity.Member;
import com.narara.superboard.reply.entity.Reply;
import com.narara.superboard.reply.interfaces.dto.ReplyCreateRequestDto;
import com.narara.superboard.reply.interfaces.dto.ReplyUpdateRequestDto;
import java.util.List;

public interface ReplyService {
    Reply createReply(Member member, ReplyCreateRequestDto replyCreateRequestDto) throws FirebaseMessagingException;
    Reply getReply(Long replyId);
    Reply updateReply(Member member, Long replyId, ReplyUpdateRequestDto replyUpdateRequestDto);
    Reply deleteReply(Member member, Long replyId);
    List<Reply> getRepliesByCardId(Long cardId);

    

}

File: Android/core/data/src/main/java/com/ssafy/data/socket/board/service/ReplyService.kt
package com.ssafy.data.socket.board.service

import com.google.gson.Gson
import com.google.gson.JsonObject
import com.ssafy.data.socket.board.model.reply.AddReplyRequestDto
import com.ssafy.data.socket.board.model.reply.DeleteReplyRequestDto
import com.ssafy.data.socket.board.model.reply.EditReplyRequestDto
import com.ssafy.database.dao.ReplyDao
import com.ssafy.database.dto.ReplyEntity
import com.ssafy.model.with.DataStatus
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ReplyService @Inject constructor(
    private val replyDao: ReplyDao,
    private val gson: Gson
) {
    suspend fun addReply(data: JsonObject) {
        val dto = gson.fromJson(data, AddReplyRequestDto::class.java)
        replyDao.insertReply(
            ReplyEntity(
                id = dto.replyId,
                cardId = dto.cardId,
                memberId = dto.memberId,
                content = dto.content,
                createAt = dto.createdAt,
                updateAt = dto.updatedAt,
            )
        )
    }

    suspend fun editReply(data: JsonObject) {
        val dto = gson.fromJson(data, EditReplyRequestDto::class.java)
        val before = replyDao.getReply(dto.replyId) ?: throw Exception("존재하지 않는 댓글 입니다.")
        replyDao.updateReply(
            before.copy(
                content = dto.content,
                updateAt = dto.updatedAt,
                isStatus = DataStatus.STAY,
            )
        )
    }

    suspend fun deleteReply(data: JsonObject) {
        val dto = gson.fromJson(data, DeleteReplyRequestDto::class.java)
        replyDao.deleteReply(dto.replyId)
    }
}